### ISSUE TITLE
Unable to prevent normal Save logic via onBeforeSave: return false due to "Name is required" error.

### ISSUE THREAD

**Issue by MediocreFantasy** [scaleflex/filerobot-image-editor#309]
Good afternoon,

In our MVC site we are using FileRobot to edit images and POST the edited image's base64 string to a controller that handles saving the data. Due to request truncation, we cannot POST a file that a user chooses to save as .png, and we do not use user-inputted file names in our storage system.

We would like to prevent the default saving behavior via the config's "onBeforeSave: ()=>{return false}" option as shown in the documentation, but this results in a "Name is required" error popup. We are trying to use the config's defaultSavedImageName option to prevent this error, but cannot seem to get the defaultSavedImageName option to do anything (with or without the onBeforeSave logic).

Below is our initialization function, which is called on $(document).ready(). If beneficial we can also provide our HTML, SaveEditedImage(), etc. Thank you in advance for your assistance.

`function OpenImageEditor(imageSource, reportPhotoId) {
    //Generates an HTML element that loads in the target image.
    var RefImage = new Image();
    RefImage.src = imageSource;

    
    var { TABS, TOOLS } = FilerobotImageEditor;
    var config = {
        source: RefImage,
        onBeforeSave: () => { return false; }, //to attempt to disable the Save window popup
        onSave: (editedImageObject, designState) =>
            SaveEditedImage(editedImageObject, reportPhotoId, designState),
        annotationsCommon: {
            fill: '#ff0000',
        },
        Text: { text: 'Your Text Here' },
        Rotate: { angle: 90, componentType: 'slider' },
        Crop: {
            presetsItems: [
                {
                    titleKey: 'classicTv',
                    descriptionKey: '4:3',
                    ratio: 4 / 3
                },
                {
                    titleKey: 'cinemascope',
                    descriptionKey: '21:9',
                    ratio: 21 / 9
                },
            ]
        },
        tabsIds: [TABS.ADJUST, TABS.ANNOTATE, TABS.WATERMARK],
        defaultTabId: TABS.ANNOTATE,
        defaultToolId: TOOLS.PEN,
        defaultSavedImageType: 'jpg', //POSTing as PNG is too long, the base64 string gets truncated.
        defaultSavedImageName: 'ThisDataNotUsed', //We have our own naming system and do not use the information inputted here.
        closeAfterSave: true
    };

    // Sticks it into the div with id="divPhotoEdit"
    var filerobotImageEditor = new FilerobotImageEditor(
        document.querySelector('#divPhotoEdit'),
        config,
    );

    filerobotImageEditor.render({
        onClose: (closingReason) => {
            filerobotImageEditor.terminate();
            console.log(closingReason);
        },
    });
}`

**Comment by AhmeeedMostafa** [comment]
Hey @MediocreFantasy I have checked ur issue and couldn't re-produce it and the image downloads fine , please provide some codesandbox produces ur issue

**Comment by MediocreFantasy** [comment]
I have written a working example of my issue in Code Sandbox.

In index.js please note the onBeforeSave return false on line 18 and the defaultSavedImageName on line 44. Expected response to clicking top-left Save button is "Data sent to server!" logged in console. Actual result is "Name is required" error despite defaultname being set.

https://codesandbox.io/s/default-name-not-working-vmbwf9?file=/index.html

**Comment by AhmeeedMostafa** [comment]
@MediocreFantasy yes u are right, the issue is produced in case of providing HTMLImageElement to `source` property not image's url,
it's fixed but waiting to be released... also u could tmp. fix it by assigning `.name` property for ur image's element and will work fine.

tmp fix from ur side, till releasing our fix:
`RefImage.name = 'UnNeededName'`

**Comment by MediocreFantasy** [comment]
@AhmeeedMostafa , that temp fix works like a charm for our application, thank you for your assistance and have a great day!

**Comment by AhmeeedMostafa** [comment]
@lee-1980 please create a new issue with a codesandbox example for ur issue to check it, ur mentioned issue is not related with the current opened one.

**Comment by AhmeeedMostafa** [comment]
Issue is fixed in the new release v4.4.0, please let us know if u still face it or not.
